### DISCUSSION TITLE
How to use [StringSyntax("Route")] without referencing AspNetCore.App?

### DISCUSSION THREAD

**Discussion by ghord** [dotnet/aspnetcore#54998] [Q&A]
I've created a library that supports the same set of routes as `AspNetCore`. I would like for my library to take advantage of `[StringSyntax('Route")]` syntax highlighting in Visual Studio, without referencing whole `AspNetCore` framewrok.

If I reference `AspNetCore.App` framework in my project (via `FrameworkReference`), everything works fine and routes are highlighted correctly.

If I do not reference `AspNetCore.App` framework, `[StringSyntax("Route")]` attribute does not work (strings are not syntax-higlighted).

I've read some mentions that this code is placed in `Microsoft.CodeAnalysis.ExternalAccess.AspNetCore`, but I cannot find nuget with this dll that I could reference as an Analyzer. Is there something I can do to make it work?

**Comment by martincostello** [comment]
My understanding is that if you can't reference the attribute through a reference from another assembly, you can create your own copy of the attribute and it should work as the attributes are found/resolved by name, not the identity of where the type is defined.

[Example copy](https://github.com/justeattakeaway/httpclient-interception/blob/main/src/HttpClientInterception/StringSyntaxAttribute.cs) so syntax highlighting works for TFMs that don't have the types defined in the framework itself.

**Reply by ghord** [reply]
The attribute is present and compiles fine, the issue is that its `"Route"` value is ignored:

![obraz](https://github.com/dotnet/aspnetcore/assets/10825459/d8fe1e32-c511-4e8c-aae3-1d7bc2031424)

**Reply by martincostello** [reply]
What about if you rename the parameter from `t` to `test` so the names match?

**Reply by ghord** [reply]
The name of the parameter does not change anything here. Here is how it looks once you add following line to the project:

```xml
<FrameworkReference Include="Microsoft.AspNetCore.App" />
```

![obraz](https://github.com/dotnet/aspnetcore/assets/10825459/91fd1ace-3dab-436e-97ad-60b855528627)

**Reply by Skleni** [reply]
@ghord did you find a solution? I'm facing the same problem.

**Reply by ghord** [reply]
No, from my research it is provided by analyzer that comes with ASP.Net SDK and relies on internal Roslyn APIs, so no easy way to achieve this.

I think it should be possible to somehow use those apis using some trickery with `IgnoreAccessChecksTo` attribute and Reflection, but I didn't try.
